src/libraries/System.Runtime.InteropServices/ref/System.Runtime.InteropServices.cs

```diff
@@ -2195,7 +2195,6 @@ public sealed partial class NativeMarshallingAttribute : System.Attribute
     {
         public Utf16StringMarshaller(string? str) { }
         public Utf16StringMarshaller(string? str, System.Span<ushort> buffer) { }
-        public ref ushort GetPinnableReference() { throw null; }
         public ushort* ToNativeValue() { throw null; }
```

Comment:
I do not see any actual uses of this GetPinnableReference method. Is it fine to delete it?

Reply:
Should the constructor that takes `System.Span<ushort> buffer` be deleted as well?

Reply:
This was required to adhere to the pattern for marshallers. I think it makes sense to force the pattern on all marshallers.

Reply:
These methods are optional. As far as I can tell, everything works just fine when they are deleted.

Reply:
I have pushed a commit to delete the constructor that takes the buffer to see whether everything still works fine.

Reply:
Yeah, they are optional. The constructor with `Span` is for the `CallerAllocatedBuffer` feature to opt-in to the potential optimization. The generator pins as the optimization instead of using `Utf16StringMarshaller`, so it never actually uses that constructor.

We included it for the consistency that the public marshallers all use this feature. I guess realistically anyone that cares about the potential optimization of not allocating would just pin as a better optimization, so having it just makes the implementation more complicated/bigger.